metagpt/team.py

```diff
@@ -30,33 +34,38 @@ class Config:
 
     def hire(self, roles: list[Role]):
         """Hire roles to cooperate"""
-        self.environment.add_roles(roles)
+        self.env.add_roles(roles)
 
     def invest(self, investment: float):
         """Invest company. raise NoMoneyException when exceed max_budget."""
         self.investment = investment
         CONFIG.max_budget = investment
-        logger.info(f'Investment: ${investment}.')
+        logger.info(f"Investment: ${investment}.")
 
     def _check_balance(self):
         if CONFIG.total_cost > CONFIG.max_budget:
-            raise NoMoneyException(CONFIG.total_cost, f'Insufficient funds: {CONFIG.max_budget}')
+            raise NoMoneyException(CONFIG.total_cost, f"Insufficient funds: {CONFIG.max_budget}")
 
-    def start_project(self, idea, send_to: str = ""):
-        """Start a project from publishing boss requirement."""
+    def run_project(self, idea, send_to: str = ""):
+        """Start a project from publishing user requirement."""
         self.idea = idea
-        self.environment.publish_message(Message(role="Human", content=idea, cause_by=BossRequirement, send_to=send_to))
+
+        # Human requirement.
+        self.env.publish_message(
+            Message(role="Human", content=idea, cause_by=UserRequirement, send_to=send_to or MESSAGE_ROUTE_TO_ALL)
+        )
 
     def _save(self):
-        logger.info(self.json())
+        logger.info(self.json(ensure_ascii=False))
 
     async def run(self, n_round=3):
         """Run company until target round or no money"""
         while n_round > 0:
             # self._save()
             n_round -= 1
-            logger.debug(f"{n_round=}")
+            logger.debug(f"max {n_round=} left.")
```

Comment:
suggest `logger.debug(f"max {n_round=} left.")` before `n_round -= 1`, or it will print max n_round=0 left, but still to run the last round. It will make a mistake.

Reply:
"xx left" is correct

Reply:
what i mean is that n_round=0 means it won't run more round, but there still a last round to run now.